components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/ImpersonatedUser.java

```
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.framework.model;
+
+import org.wso2.carbon.identity.application.authentication.framework.exception.UserIdNotFoundException;
+
+/**
+ * ImpersonatedUser is the class that represents the impersonated subject.
+ */
+public class ImpersonatedUser extends AuthenticatedUser {
+
+    public ImpersonatedUser() {
+
+        super();
+    }
+
+    public ImpersonatedUser(AuthenticatedUser authenticatedUser) throws UserIdNotFoundException {
+
+        super(authenticatedUser);
+    }
+
+    @Override
+    public ImpersonatedUser getImpersonatedUser() {
+
+        return null;
```

Lets throw an exception here as well.

---

addressed with 2af5ac045b6323d153825d3712f2801cd836ddd4